ecoscope/base/relocations.py

```diff
@@ -0,0 +1,215 @@
+    RelocsSpeedFilter,
+)
+from ecoscope.base.straightrack import StraightTrackProperties
+from ecoscope.base import EcoDataFrame
+
+
+class Relocations(EcoDataFrame):
+    """
+    Relocation is a model for a set of fixes from a given subject.
+    Because fixes are temporal, they can be ordered asc or desc. The additional_data dict can contain info
+    specific to the subject and relocations: name, type, region, sex etc. These values are applicable to all
+    fixes in the relocations array. If they vary, then they should be put into each fix's additional_data dict.
+    """
+
+    @classmethod
+    def from_gdf(cls, gdf, groupby_col=None, time_col="fixtime", uuid_col=None, **kwargs):
+        """
+        Parameters
+        ----------
+        gdf : GeoDataFrame
+            Observations data
+        groupby_col : str, optional
+            Name of `gdf` column of identities to treat as separate individuals. Usually `subject_id`. Default is
+            treating the gdf as being of a single individual.
+        time_col : str, optional
+            Name of `gdf` column containing relocation times. Default is 'fixtime'.
+        uuid_col : str, optional
+            Name of `gdf` column of row identities. Used as index. Default is existing index.
+        """
+
+        assert {"geometry", time_col}.issubset(gdf)
+
+        if kwargs.get("copy") is not False:
+            gdf = gdf.copy()
+
+        if groupby_col is None:
+            if "groupby_col" not in gdf:
+                gdf["groupby_col"] = 0
+        else:
+            gdf["groupby_col"] = gdf.loc[:, groupby_col]
+
+        if time_col != "fixtime":
+            gdf["fixtime"] = gdf.loc[:, time_col]
+
+        if not pd.api.types.is_datetime64_any_dtype(gdf["fixtime"]):
+            warnings.warn(
+                f"{time_col} is not of type datetime64. Attempting to automatically infer format and timezone. "
+                "Results may be incorrect."
+            )
+            gdf["fixtime"] = pd.to_datetime(gdf["fixtime"])
+
+        if gdf["fixtime"].dt.tz is None:
+            warnings.warn(f"{time_col} is not timezone aware. Assuming datetime are in UTC.")
+            gdf["fixtime"] = gdf["fixtime"].dt.tz_localize(tz="UTC")
+
+        if gdf.crs is None:
+            warnings.warn("CRS was not set. Assuming geometries are in WGS84.")
+            gdf.set_crs(4326, inplace=True)
+
+        if uuid_col is not None:
+            gdf.set_index(uuid_col, drop=False, inplace=True)
+
+        gdf["junk_status"] = False
+
+        default_cols = ["groupby_col", "fixtime", "junk_status", "geometry"]
+        extra_cols = gdf.columns.difference(default_cols)
+        extra_cols = extra_cols[~extra_cols.str.startswith("extra__")]
+
+        assert gdf.columns.intersection("extra__" + extra_cols).empty, "Column names overlap with existing `extra`"
+
+        gdf.rename(columns=dict(zip(extra_cols, "extra__" + extra_cols)), inplace=True)
+
+        return Relocations(gdf=gdf)
```

Comment:
```suggestion
        return cls(gdf=gdf)
```
Do we want people to subclass this? Probably not. Might it happen? Sure. So I think this is advisable. Cf. https://stackoverflow.com/questions/46128877/should-i-use-the-class-name-or-cls-param-to-construct-a-an-instance